pco/src/mode.rs

```diff
@@ -103,11 +103,10 @@ impl<L: Latent> Mode<L> {
   }
 }
 
-#[derive(Clone, Copy, Debug, Default, PartialEq)]
-pub enum Bid<L: Latent> {
-  /// "Don't use this mode" (TODO: elaborate)
-  #[default]
-  Forfeit,
-  /// Bid to use this mode (TODO: elaborate)
-  Candidate { mode: Mode<L>, bits_saved_per_num: f64 },
+pub(crate) struct Bid<T: NumberLike> {
+  pub mode: Mode<T::L>,
+  pub bits_saved_per_num: f64,
+  // we include a split_fn since modes like FloatMult can benefit from extra
+  // information (inv_base) not captured entirely in the mode
+  pub split_fn: Box<dyn FnOnce(&[T]) -> Vec<Vec<T::L>>>,
```

Comment:
This makes sense for how it's being used right now.

I'm wondering if there is an implicit API change/refinement that should be clarified somewhere.  In particular, as a user of the library I would wonder: if `split_latents` receives side info besides the `Mode`, then what do I need to know about that side info in order to use the library correctly?

I think the answer to the above is: users do not need to know anything about the side info; it is an internal detail of the compression algorithm that allows it to auto-detect parameters more effectively and, in particular, the decompression step does not need to know anything about it.

Come to think of it, maybe the place to doc that is right here.
```suggestion
  // we include a split_fn since modes like FloatMult can benefit from extra
  // information (inv_base) not captured entirely in the mode.  This extra
  // information is an implementation detail of the compressor, not part of the
  // format itself, and is not / does not need to be known to the decompressor.
  pub split_fn: Box<dyn FnOnce(&[T]) -> Vec<Vec<T::L>>>,
```

Reply:
Oh I absolutely do not intend to expose this to users (even the power users who might consider looking at `NumberLike`. It's only `pub(crate)`.

Reply:
Ah yeah, I meant to say users and maintainers -- imo it's worth saying this here so that readers don't have to think hard to know it.  Is there reason not to include it?